src/main/java/core/basesyntax/Square.java

```diff
@@ -0,0 +1,23 @@
+package core.basesyntax;
+
+public class Square extends Figure {
+    private int side;
+    private String color = getColor();
+
+    public Square(int side, String color) {
+        this.side = side;
+        this.color = color;
+    }
+
+    @Override
+    public double areaCalculator() {
+        return (side * side);
+    }
+
+    @Override
+    public void draw() {
+        System.out.println("Figure square, area: " + areaCalculator()
+                + ", side: " + side
+                + ", color: " + color);
```

Comment:
The 'draw' method is printing directly to the console, which is not a good practice for a method that's supposed to represent a behavior of the object. Consider refactoring this to return a String representation of the object, and then print that String in the main application if needed.